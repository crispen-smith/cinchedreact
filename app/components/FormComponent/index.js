/**
 *
 * FormComponent
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const FC = styled.form`
  border-radius: 10px;
  background-color: rgba(255, 255, 255, 0.1);
  font-size: 1.25rem;

  .form-row-primary {
    padding: 1rem;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    text-align: center;
  }
`;

/* eslint-disable react/prefer-stateless-function */
class FormComponent extends React.PureComponent {
  render() {
    return <FC onSubmit={this.props.onSubmit}>{this.props.children}</FC>;
  }
}

FormComponent.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]).isRequired,
};

export default FormComponent;
