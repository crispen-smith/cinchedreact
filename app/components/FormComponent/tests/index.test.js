import React from 'react';
import renderer from 'react-test-renderer';

import FormComponent from '../index';

describe('<NameBox />', () => {
  const onSubmit = () => 1;

  it('Renders a FormComponent', () => {
    const tree = renderer
      .create(<FormComponent onSubmit={onSubmit} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
