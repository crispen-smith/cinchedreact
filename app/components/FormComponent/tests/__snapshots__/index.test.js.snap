// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NameBox /> Renders a FormComponent 1`] = `
<form
  className="FormComponent__FC-jRyaDy hZINgg"
  onSubmit={[Function]}
/>
`;
